homeassistant/components/simplefin/config_flow.py

```diff
@@ -66,10 +84,32 @@ async def async_step_user(
 
         return self.async_show_form(
             step_id="user",
-            data_schema=vol.Schema(
-                {
-                    vol.Required(CONF_ACCESS_URL): str,
-                }
-            ),
+            data_schema=SCHEMA,
             errors=errors,
         )
+
+    async def async_step_reauth(
+        self, user_input: Mapping[str, Any]
+    ) -> ConfigFlowResult:
+        """Perform reauth upon an API authentication error."""
+        self.reauth_entry = self.hass.config_entries.async_get_entry(
+            self.context["entry_id"]
+        )
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Dialog that informs the user that reauth is required."""
+        if user_input is not None:
+            access_url, errors = await _validate_input(user_input)
+            if not errors and self.reauth_entry:
```

Comment:
Shouldn't we check here too if another entry is using the user supplied access URL, but instead of aborting the flow, like we do in the user flow, we ask the user to fill in the form again?